internal/cli/atlas/backup/compliancepolicy/policies/update/update.go

```diff
@@ -0,0 +1,163 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package update
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	store "github.com/mongodb/mongodb-atlas-cli/internal/store/atlas"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+type combinedStore interface {
+	store.CompliancePolicyItemUpdater
+	store.ProjectLister
+	store.CompliancePolicyDescriber
+}
+
+type Opts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	store  combinedStore
+	fs     afero.Fs
+	path   string
+	policy *atlasv2.DataProtectionSettings
+}
+
+const (
+	active = "ACTIVE"
+)
+
+var updateTemplate = `Your backup compliance policy is being updated with the following policies:
+
+ID	FREQUENCY INTERVAL	FREQUENCY TYPE	RETENTION
+{{- range .ScheduledPolicyItems}}
+{{.Id}}	{{if eq .FrequencyType "hourly"}}{{.FrequencyInterval}}{{else}}-{{end}}	{{.FrequencyType}}	{{.RetentionValue}} {{.RetentionUnit}}
+{{- end}}
+{{if .OnDemandPolicyItem}}{{.OnDemandPolicyItem.Id}}	-	{{.OnDemandPolicyItem.FrequencyType}}	{{.OnDemandPolicyItem.RetentionValue}} {{.OnDemandPolicyItem.RetentionUnit}}{{end}}
+`
+var updateWatchTemplate = `Your backup compliance policy has been updated with the following policies:
+
+ID	FREQUENCY INTERVAL	FREQUENCY TYPE	RETENTION
+{{- range .ScheduledPolicyItems}}
+{{.Id}}	{{if eq .FrequencyType "hourly"}}{{.FrequencyInterval}}{{else}}-{{end}}	{{.FrequencyType}}	{{.RetentionValue}} {{.RetentionUnit}}
+{{- end}}
+{{if .OnDemandPolicyItem}}{{.OnDemandPolicyItem.Id}}	-	{{.OnDemandPolicyItem.FrequencyType}}	{{.OnDemandPolicyItem.RetentionValue}} {{.OnDemandPolicyItem.RetentionUnit}}{{end}}
+`
+var example = `How to run "atlas backups compliancepolicy policies update" with --file.
+As an example, the file should be in the following format:
+
+{
+	"frequencyInterval": 1,
+	"frequencyType": "daily",
+	"id": "stringstringstringstring",
+	"retentionUnit": "days",
+	"retentionValue": 0
+}
+
+To get the ID of a policy item, run "atlas backups compliancepolicy policies describe".
+
+Finally, run the command as such: "atlas backups compliancepolicy policies update --file /path/to/file"
+`
+
+var errorCode500Template = `received an internal error on the server side, but we would encourage you to double check your inputs.
+For this command, invalid inputs are known to cause internal errors in some situations`
+
```

Comment:
Not entirely sure if this should be included, but the message is true. If the idea is OK, I'm going to double check the wording of it with the docs team.

Reply:
why is the 500? is the json formatted wrongly? missing fields? incompatible fields?